src/GraphQL/Utilities/SchemaBuilder.cs

```diff
@@ -350,7 +350,7 @@ protected virtual UnionGraphType ToUnionType(GraphQLUnionTypeDefinition unionDef
 
             CopyMetadata(type, typeConfig);
 
-            var possibleTypes = unionDef.Types.Select(x => GetType(x.Name.Value));
+            var possibleTypes = unionDef.Types.Select(x => GetType(x.Name.Value) ?? new GraphQLTypeReference(x.Name.Value));
```

Comment:
I was guided by [this behavior](https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Utilities/SchemaBuilder.cs#L466).